zkvm/src/signature/prover.rs

```diff
@@ -0,0 +1,189 @@
+#![allow(non_snake_case)]
+
+use crate::errors::VMError;
+use crate::signature::multikey::Multikey;
+use crate::signature::musig::*;
+use crate::signature::VerificationKey;
+use crate::transcript::TranscriptProtocol;
+use curve25519_dalek::constants::RISTRETTO_BASEPOINT_POINT;
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+use rand;
+use subtle::ConstantTimeEq;
+
+#[derive(Copy, Clone)]
+pub struct NoncePrecommitment([u8; 32]);
+
+#[derive(Copy, Clone, Debug)]
+pub struct NonceCommitment(RistrettoPoint);
+
+pub struct PartyAwaitingPrecommitments {
+    transcript: Transcript,
+    multikey: Multikey,
+    x_i: Scalar,
+    r_i: Scalar,
+    R_i: NonceCommitment,
+}
+
+pub struct PartyAwaitingCommitments {
+    transcript: Transcript,
+    multikey: Multikey,
+    x_i: Scalar,
+    r_i: Scalar,
+    nonce_precommitments: Vec<NoncePrecommitment>,
+}
+
+pub struct PartyAwaitingSiglets {
+    multikey: Multikey,
+    c: Scalar,
+    nonce_commitments: Vec<NonceCommitment>,
+}
+
+impl NonceCommitment {
+    fn precommit(&self) -> NoncePrecommitment {
+        let mut h = Transcript::new(b"MuSig.nonce-precommit");
+        h.commit_point(b"R_i", &self.0.compress());
```

Comment:
```suggestion
        h.commit_point(b"R", &self.0.compress());
```

I generally want to simplify the labels and naming so we don't cargocult quite arbitrary notation from the paper.

Reply:
Especially since the paper can be rewritten, but labels must stay forever for compatibility.

Reply:
In my mind, the thing we are committing is the "per-party" `R` (and hence, called `R_i`)... whereas the "aggregated" `R` is the one that should really be called `R`.

Anyway, not a big deal & I changed it.